Neos.Media.Browser/Classes/Controller/AssetController.php

```diff
@@ -379,22 +316,89 @@ public function replaceAssetResourceAction(Asset $asset)
     }
 
     /**
-     * Edit an asset
+     * Show an asset
      *
-     * @param Asset $asset
+     * @param string $assetSourceIdentifier
+     * @param string $assetProxyIdentifier
      * @return void
+     * @throws StopActionException
+     * @throws UnsupportedRequestTypeException
      */
-    public function editAction(Asset $asset)
+    public function showAction(string $assetSourceIdentifier, string $assetProxyIdentifier)
     {
-        $this->view->assignMultiple([
-            'tags' => $asset->getAssetCollections()->count() > 0 ? $this->tagRepository->findByAssetCollections($asset->getAssetCollections()->toArray()) : $this->tagRepository->findAll(),
-            'asset' => $asset,
+        if (!isset($this->assetSources[$assetSourceIdentifier])) {
+            throw new \RuntimeException('Given asset source is not configured.', 1509702178632);
+        }
+
+        $assetBrowser = $this->assetSources[$assetSourceIdentifier]->getAssetProxyRepository();
+        try {
+            $assetProxy = $assetBrowser->getAssetProxy($assetProxyIdentifier);
+
+            $this->view->assignMultiple([
+//            'tags' => $assetProxy->getAssetCollections()->count() > 0 ? $this->tagRepository->findByAssetCollections($assetProxy->getAssetCollections()->toArray()) : $this->tagRepository->findAll(),
+                'assetProxy' => $assetProxy,
             'assetCollections' => $this->assetCollectionRepository->findAll()
-        ]);
+            ]);
+        } catch (AssetNotFoundException $e) {
+            $this->throwStatus(404, 'Asset not found');
+        } catch (AssetSourceConnectionException $e) {
+            $this->view->assign('connectionError', $e);
+        }
     }
 
     /**
+     * Edit an asset
+     *
+     * @param string $assetSourceIdentifier
+     * @param string $assetProxyIdentifier
      * @return void
+     * @throws StopActionException
+     * @throws UnsupportedRequestTypeException
+     */
+    public function editAction(string $assetSourceIdentifier, string $assetProxyIdentifier)
+    {
+        if (!isset($this->assetSources[$assetSourceIdentifier])) {
+            throw new \RuntimeException('Given asset source is not configured.', 1509632166496);
+        }
+
+        $assetSource =  $this->assetSources[$assetSourceIdentifier];
+        $assetProxyRepository = $assetSource->getAssetProxyRepository();
+
+        try {
+            $assetProxy = $assetProxyRepository->getAssetProxy($assetProxyIdentifier);
+
+            $tags = [];
+            if ($assetProxyRepository instanceof SupportsTagging) {
+                if ($assetProxyRepository instanceof SupportsCollections) {
```

Comment:
Seems this inner if could be added to the condition of the one above.

Reply:
✅